### PR TITLE
Feat/623/route api

### PR DESCRIPTION
**Pull request by JasonFritsche**
#623

### CODE REVIEW

**Comment by JasonFritsche** [comment]
@raae I think I've got the pages (routes, region, subregion) working (big thanks to @jtfairbank ) using the filesystem route api. Could you please take a look when you're able and let me know if I should change/refactor anything. In the meantime I will work on getting rid of the templates and migrating over the templates to the new pages. Thank you!

File: src/pages/regions/index.tsx [hunk 1]
```diff
@@ -96,8 +97,11 @@ const RegionsPage: FC<Props> = ({
 
 export default RegionsPage
 
-export const pageQuery = graphql`
-  query RegionsQuery {
+export const query = graphql`
+  query MyQuery($id: String) {
```
Comment (raae) [inline src/pages/regions/index.tsx:96]: suggestion: markdown query not in use, can be removed

File: src/pages/regions/index.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ const RegionsPage: FC<Props> = ({
     regions: { nodes: regions },
   },
 }) => {
+  console.log('dataatatatata', regions)
```
Comment (raae) [inline src/pages/regions/index.tsx]: nitpick: remove logging

File: src/pages/regions/index.tsx [hunk 3]
```diff
@@ -21,6 +21,7 @@ const RegionsPage: FC<Props> = ({
...
   const createRegionHref = (region: Region) => {
```
Comment (raae) [inline src/pages/regions/index.tsx]: suggestion: use [gatsbyPath](https://www.gatsbyjs.com/docs/reference/routing/file-system-route-api/#gatsbypath-example) approach in the query to making sure the you get correct slugified path

**Review by raae — COMMENTED** [review]
Great work, added some [conventional comments](https://conventionalcomments.org/) below.

praise: {daSubregion.region__name} folder 🥳

suggestion(non-blocking): I would set a slug on regions and subregions similar to routes equal to the filename, see the route creation to see how to set them.

File: src/pages/regions/index.tsx (hunk 2 above)
Comment (JasonFritsche) [reply src/pages/regions/index.tsx]: removed

File: src/pages/regions/index.tsx (hunk 1 above)
Comment (JasonFritsche) [reply src/pages/regions/index.tsx:96]: removed the markdown query

**Comment by jtfairbank** [comment]
Hey @JasonFritsche just a heads up that we've refactored our gatsby code to use typescript in #736.  Might need to rebase or merge those changes into here and adjust your changes to work with typescript.

**Comment by JasonFritsche** [comment]
> Hey @JasonFritsche just a heads up that we've refactored our gatsby code to use typescript in #736. Might need to rebase or merge those changes into here and adjust your changes to work with typescript.

@jtfairbank I'll merge in the latest and see what happens. Currently trying to get this slug issue resolved. I don't have time to look at this today, but will be able to spend some time tomorrow and over the weekend.

**Comment by JasonFritsche** [comment]
@raae @jtfairbank I've merged in the latest from `saga`. The e2e tests pass. I've removed the route and regions templates. I am however, unable to figure out how to create the slug for Region and Subregion. I've left my attempt at creating the slug in the draft PR.

File: gatsby-node.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const path = require('path')
```
Comment (coderbyheart) [inline gatsby-node.js]: That file should not be in here.

**Review by coderbyheart — CHANGES_REQUESTED** [review]

**Comment by jtfairbank** [comment]
Hey @JasonFritsche I can tackle the slugs today!

**Comment by JasonFritsche** [comment]
> Hey @JasonFritsche I can tackle the slugs today!

@jtfairbank I really appreciate the help. This story is definitely one that's out of my comfort zone. I have availability this evening (after 1700 CST) if you'd be interested in pair programming.

**Comment by raae** [comment]
I could join in some pairing after 19:30 tonight...

**Comment by jtfairbank** [comment]
Hey @raae unfortunately you two are in very different but similar sounding timezones. :sweat_smile: CST is 7 hours behind CEST.

@JasonFritsche I'm still traveling so it'll be easier to tackle this independently unless you want to wait until the Code Jam next Monday (7-9pm CST)?  We can always review the code together after it's merged in to create a learning opportunity!

**Comment by raae** [comment]
@JasonFritsche you want to pair at 20.00 CEST?

**Comment by JasonFritsche** [comment]
@raae unfortunately I won't be available until 1700 CST which I believe is 0300 CEST. 

@jtfairbank I don't mind if you work on this feature. There are (I believe) two outstanding issues in this ticket, the slug creation/use, and removing the gatsby-node.js file. Both of these items are presenting a challenge to me.

**Comment by raae** [comment]
Totally misunderstood the timezone issue 🤦‍♀️

**Comment by jtfairbank** [comment]
Ok y'all here's what I've done:

  - Added e2e tests for the regions index page, region pages, and subregion pages.
  - Created slugs based on the filesystem for Regions / Subregions in `/gatsby/transform-nodes.ts`
  - Attempted to use GatsbyPath to get the links for the dynamic region & subregion pages.
  
Where I'm stuck:

  - The subregions use two parameters in their dynamic path which works with the fileSystemAPI but doesn't work with GatsbyPath.  It is attempting to create paths that look like this "/regions/{DASubregion.region__slug}/serbia/" instead of "/regions/the-balkans/serbia/"

@raae any insights here?

**Comment by raae** [comment]
I have not had that use case before @jtfairbank, so no insight 😬

File: gatsby-node.ts [hunk 5]
```diff
@@ -42,9 +41,7 @@ Create Dynamic Pages
 ================================================================================
 */
 export const createPages: GatsbyNode['createPages'] = async (args) => {
-  await pages.createRegionPages(args)
-  await pages.createSubregionPages(args)
-  await pages.createRoutePages(args)
+  // Currently all pages are created using the Filesystem Route API
```
Comment (coderbyheart) [inline gatsby-node.ts]: Remove entire block, because `createPages` is not used.

**Review by coderbyheart — APPROVED** [review]
Feedback in #768

**Comment by jtfairbank** [comment]
Hey @coderbyheart! #768 looks great, we can merge it into here once it passes CI, then merge this into saga?

**Comment by coderbyheart** [comment]
Will do!

**Comment by coderbyheart** [comment]
Thanks everyone! 🚀

**Comment by jtfairbank** [comment]
Thank you for carrying this one home @coderbyheart!

**Pull request merged**